paddle/phi/kernels/onednn/squeeze_kernel.cc

```diff
@@ -60,6 +60,12 @@ void SqueezeInferKernel(const Context& dev_ctx,
                         DenseTensor* out) {
   auto x_dims = x.dims();
   std::vector<int32_t> tmp(axes.GetData().begin(), axes.GetData().end());
+
+  // Currently there is only transformation for tensors, while attr axes still
+  // follows default dtype instead of oneDNN dtype, so here manually change it
+  // TODO(Li Xinyi): Is there a way to achieve it at earlier stage?
+  funcs::GetSqueezeNewAxes(tmp);
```

Comment:
``` 
 std::vector<int64_t> formated_axis = axes.GetData();
  if ((x_dims.size() >= 3) &&
      (phi::OneDNNContext::tls().get_cur_paddle_data_layout() ==
       phi::DataLayout::kNHWC)) {
    int axis_size = axes.GetData().size();
    std::vector<int> count(axis_size, 0);
    for (int i = 0; i < axis_size; i++) {
      if (axes.GetData()[i] < 0) {
        formated_axis[i] = axes.GetData()[i] + axis_size;
      }
    }
  }
  std::vector<int32_t> tmp(formated_axis.begin(), formated_axis.end());
```
It seems can use such method to solve it?

Reply:
> ```
>  std::vector<int64_t> formated_axis = axes.GetData();
>   if ((x_dims.size() >= 3) &&
>       (phi::OneDNNContext::tls().get_cur_paddle_data_layout() ==
>        phi::DataLayout::kNHWC)) {
>     int axis_size = axes.GetData().size();
>     for (int i = 0; i < axis_size; i++) {
>       if (axes.GetData()[i] < 0) {
>         formated_axis[i] = axes.GetData()[i] + axis_size;
>       }
>     }
>   }
>   std::vector<int32_t> tmp(formated_axis.begin(), formated_axis.end());
> ```
> 
> It seems can use such method to solve it?

Thanks for your advice! Do you think it's better to add check function when `get_cur_paddle_data_layout() ==
       phi::DataLayout::NDHWC`?

Reply:
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/phi/kernels/transfer_layout_kernel.cc#L161, maybe still need to kNHWC or NHWC? (kNHWC define equals NHWC)

Reply:
> https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/phi/kernels/transfer_layout_kernel.cc#L161, maybe still need to kNHWC or NHWC? (kNHWC define equals NHWC)

Seems it assumed the dims would only be 4...  Here indicates 5D dims. So maybe add one more case is also good?
https://github.com/PaddlePaddle/Paddle/blob/9d7150b8747997d5a2264c834376ed5f55a4c9d4/paddle/common/layout.h#L36

Reply:
sure, I think it's more reasonable